File: javascript/22-names-scores.js
var fs = require("fs");

function getNameScore(name) {
  let score = 0;
  for (let i = 0; i < name.length; i++) {
    score += name.charCodeAt(i) - 64;
  }

  return score;
}

function namesScores(arr) {
  arr.sort();

  let score = 0;
  for (let i = 0; i < arr.length; i++) {
    score += getNameScore(arr[i]) * (i + 1);
  }

  return score;
}

const test = fs
  .readFileSync("p022_names.txt")
  .toString()
  .split(",")
  .map(name => name.replace(/"/g, ""));

console.log(namesScores(test));

File: go/22-names-scores.go
package main

import "io/ioutil"
import "fmt"
import "strings"
import "sort"

func getNameScore(name string) int {
	score := 0
	for i := 0; i < len(name); i++ {
		score += (int(name[i]) - 64)
	}

	return score
}

func namesScores(arr []string) int {
	sort.Strings(arr)

	score := 0
	for i := 0; i < len(arr); i++ {
		score += getNameScore(arr[i]) * (i + 1)
	}

	return score
}

func main() {
	data, _ := ioutil.ReadFile("p022_names.txt")

	test := strings.Split(string(data), ",")
	for i := 0; i < len(test); i++ {
		test[i] = strings.ReplaceAll(test[i], "\"", "")
	}

	fmt.Println(namesScores(test))
}
